### ISSUE TITLE
BUG: Wrong version printed by hyprkeys version

### ISSUE THREAD

**Issue by bloominstrong** [hyprland-community/Hyprkeys#13]
Calling `hyprkeys version` gives `Hyprkeys: v1.1.0` which has been set in [cmd/version.go](https://github.com/hyprland-community/Hyprkeys/blob/0dac4d64fd777072f8bfc9563af23d0a0903a3fe/cmd/version.go#L9) in release v1.0.2

It could be worth considering something similar to what [nebula](https://github.com/slackhq/nebula/blob/57eb80e9fbb09ac16f22b38b258a18005b7a6cb1/Makefile#L23) does in using ldflags to set the Version variable at compile if not already set. this means that the version is

**Comment by NotAShelf** [comment]
Good catch! I'll publish a fix by tomorrow, and look into using ldflags afterwards. Seems to be the standard-ish in nixpkgs anyway.

**Comment by donovanglover** [comment]
Hi @NotAShelf

I'm interested in adding Hyprkeys to nixpkgs and plan to update https://github.com/NixOS/nixpkgs/pull/241201 once the new version comes out.

I'm not sure why, but the current [archive](https://github.com/hyprland-community/Hyprkeys/archive/v1.0.2.tar.gz) returns "the given path has multiple possibilities", thus breaking the build. Maybe it's because the [tag](https://github.com/hyprland-community/Hyprkeys/activity?ref=refs/tags/v1.0.2) was deleted?

**Comment by bloominstrong** [comment]
@donovanglover, 

I think the problem is there is a branch called v1.0.2 . You can reference the tag directly with https://github.com/hyprland-community/Hyprkeys/archive/refs/tags/v1.0.2.tar.gz
But I don't know if box's fetch from GitHub function can specifically use the ref tags address instead

**Comment by donovanglover** [comment]
Thanks for the hint! PR is now fixed

**Comment by NotAShelf** [comment]
fetchFromGitHub should be able to fetch the v1.0.2 branch with `rev = "v${version$"` or similar

**Comment by NotAShelf** [comment]
p.s. I deleted the v1.0.2 branch, should be easier to do so now

**Comment by NotAShelf** [comment]
@bloominstrong I've pushed some changes that I think should solve the issue. The version string is now obtained from ldlags in both Makefile and nix builds. I've not tested goreleaser, but 
it should probably also work. Please let me know if the issue is fixed.
